core/src/idx/ft/analyzer/tokenizer.rs

```diff
@@ -84,26 +84,31 @@ impl TryFrom<Tokens> for Value {
 #[derive(Clone, Debug, PartialOrd, PartialEq, Eq, Ord, Hash)]
 pub(super) enum Token {
 	Ref {
-		chars: (Position, Position),
+		chars: (Position, Position, Position),
```

Comment:
We now store in the offsets for each token the `start` and `end` position of the original term, and the `start` position of the generated token.
Eg. `Hello` generates these tokens (lowercase + edgengram):
`He` `el` `ll`, etc..
The respective offset are now [0,0,5] and [0,1.5] and [0,2,5].

Reply:
Does this need revision at all?

Reply:
Not it doesn't. The tokens are not directly serialized. The serialisation is done manually as a vec of u32 to have the benefit of compression. The "database upgrade" test has been updated to check that everything is fine